src/request-queue.js

```diff
@@ -87,8 +87,8 @@ export class RequestQueue {
         clearTimeout(this._poller)
         const requests = this._event_queue.length > 0 ? this.formatQueue() : {}
         this._event_queue.length = 0
-        for (let url in requests) {
-            const { data, options } = requests[url]
+        for (let key in requests) {
+            const { url, data, options } = requests[key]
```

Comment:
Interesting. There are tests under src/__tests__/request-queue.js - mind adding one or editing existing ones in a way which demonstrates the problem? E.g. how was enqueue called in this case?

Reply:
Sure thing, that was the first item for today!

It's kinda unfortunate: we had a similar test for poll that works, but didn't have one for unload.